drivers/regulator/regulator_rpi_pico.c

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2023 TOKITA Hiroshi <tokita.hiroshi@fujitsu.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT raspberrypi_core_supply_regulator
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/dt-bindings/regulator/rpi_pico.h>
+#include <zephyr/sys/linear_range.h>
+#include <zephyr/toolchain.h>
+#include <hardware/regs/vreg_and_chip_reset.h>
+#include <hardware/structs/vreg_and_chip_reset.h>
+
+#define VREG_VSEL_IDX_OFFSET 6
+
+/*
+ * Registers are defined from 800000 microvolt,
+ * but since the operation becomes unstable,
+ * use from 850000 microvolt like Pico-SDK.
+ */
+static const struct linear_range core_ranges[] = {
+	LINEAR_RANGE_INIT(850000u, 50000u, 0u, 9u),
```

Comment:
I've not familiar with this chip, but it seems to me that lower voltages could be useful if the chip was underclocked or entering a low power mode. Is there no situation where the 800000uV output would be useful?

Reply:
I noticed these values are usable when externally supplying the core voltage. (RP2040 usually drives by voltage from the built-in regulator, but can configure not use the regulator.)
Thus, I changed to make all values available.
However, this does not pass the test with this board.
So I made it possible to set upper and lower test voltage limits in the test so that we could exclude unstable settings from the test.

Reply:
I think this is a good solution, thank you for enabling this feature